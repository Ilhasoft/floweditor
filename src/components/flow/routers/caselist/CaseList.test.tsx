import CaseList from 'components/flow/routers/caselist/CaseList';
import { Operators } from 'config/interfaces';
import React from 'react';
import {
  act,
  fireEvent,
  fireTembaSelect,
  fireUnnnicInputChangeText,
  fireUnnnicSelect,
  getUnnnicSelectValue,
  render,
  waitForDomChange
} from 'test/utils';
import { mock } from 'testUtils';
import * as utils from 'utils';

mock(utils, 'createUUID', utils.seededUUIDs());

const caseUUID1 = utils.createUUID();
const caseUUID2 = utils.createUUID();

const cases = [
  {
    uuid: caseUUID1,
    kase: {
      uuid: caseUUID1,
      type: Operators.has_any_word,
      arguments: ['Red, r'],
      category_uuid: utils.createUUID()
    },
    categoryName: 'Red',
    valid: true
  },
  {
    uuid: caseUUID2,
    kase: {
      uuid: caseUUID2,
      type: Operators.has_any_word,
      arguments: ['Green, g'],
      category_uuid: utils.createUUID()
    },
    categoryName: 'Green',
    valid: true
  }
];

describe(CaseList.name, () => {
  describe('render', () => {
    it('should render empty list', () => {
      const { baseElement } = render(<CaseList cases={[]} onCasesUpdated={jest.fn()} />);
      expect(baseElement).toMatchSnapshot();
    });

    it('should render list of cases', () => {
      const { baseElement } = render(<CaseList cases={cases} onCasesUpdated={jest.fn()} />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('updates', () => {
    it('should remove cases', () => {
      const { baseElement, getByTestId, queryByTestId } = render(
        <CaseList cases={cases} onCasesUpdated={jest.fn()} />
      );

      const removeCase = `remove-case-${cases[0].uuid}`;
      fireEvent.click(getByTestId(removeCase));
      expect(baseElement).toMatchSnapshot();

      // our case should be gone
      expect(queryByTestId(removeCase)).toBeNull();
    });

    it('should update cases', async () => {
      const onCasesUpdated = jest.fn();
      const { baseElement, getAllByTestId } = render(
        <CaseList cases={cases} onCasesUpdated={onCasesUpdated} />
      );

      const argsInput = getAllByTestId('arguments')[0] as any;
      await act(async () => {
        fireUnnnicInputChangeText(argsInput, 'Purple, p');
      });

      expect(onCasesUpdated).toHaveBeenCalledTimes(5);
      expect(baseElement).toMatchSnapshot();
    });

    it('changes default operator on add', async () => {
      const onCasesUpdated = jest.fn();

      // start with an empty list
      const { baseElement, getAllByTestId } = render(
        <CaseList cases={[]} onCasesUpdated={onCasesUpdated} />
      );

      // we should have one operator for the default
      expect(getAllByTestId('temba_select_operator').length).toBe(1);

      // switch default to greater than operator and populate it
      fireUnnnicSelect(
        getAllByTestId('temba_select_operator')[0],
        { value: [{ type: Operators.has_number_gt }] },
        'value'
      );
      const argsInput = getAllByTestId('arguments')[0];
      await act(async () => {
        fireUnnnicInputChangeText(argsInput, '42');
      });

      // now we should have two rules
      expect(getAllByTestId('temba_select_operator').length).toBe(2);

      // our new default operator should match the last one we entered
      const newOperator = getAllByTestId('temba_select_operator')[1];
      expect(getUnnnicSelectValue(newOperator)).toBe('has_number_gt');

      expect(baseElement).toMatchSnapshot();
    });
  });
});
