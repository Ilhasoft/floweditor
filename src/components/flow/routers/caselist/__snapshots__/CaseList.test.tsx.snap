// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaseList render should render empty list 1`] = `
<body>
  <div>
    <div
      class="case_list_container "
    >
      <div
        class="case_list"
      >
        <div
          class="kase case_list_case"
        >
          <div
            class="ele group"
          >
            <div
              class="kase has_any_word"
              data-draggable="true"
            >
              <span
                class="fe-chevrons-expand dnd_icon"
                data-draggable="true"
              />
              <div
                class="choice"
              >
                <div
                  class="small "
                >
                  <temba-select
                    data-testid="temba_select_operator"
                    errors="[]"
                    name="operator"
                    namekey="verboseName"
                    valuekey="type"
                    values="[{\\"type\\":\\"has_any_word\\",\\"verboseName\\":\\"has any of the words\\",\\"operands\\":1,\\"visibility\\":[\\"message\\",\\"message_offline\\"]}]"
                  />
                </div>
              </div>
              <div
                class="single_operand"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="-1"
                      name="arguments"
                      placeholder=""
                      type="text"
                      value=""
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="categorize_as"
                data-draggable="true"
              >
                categorize as
              </div>
              <div
                class="category"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="36"
                      name="Exit Name"
                      type="text"
                      value=""
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="fe-x remove_icon"
                data-testid="remove-case-477ac8b4-25e2-483a-8686-2d1332c4da1c"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`CaseList render should render list of cases 1`] = `
<body>
  <div>
    <div
      class="case_list_container "
    >
      <div
        class="case_list"
      >
        <div
          class="kase case_list_case"
        >
          <div
            class="ele group"
          >
            <div
              class="kase has_any_word"
              data-draggable="true"
            >
              <span
                class="fe-chevrons-expand dnd_icon"
                data-draggable="true"
              />
              <div
                class="choice"
              >
                <div
                  class="small "
                >
                  <temba-select
                    data-testid="temba_select_operator"
                    errors="[]"
                    name="operator"
                    namekey="verboseName"
                    valuekey="type"
                    values="[{\\"type\\":\\"has_any_word\\",\\"verboseName\\":\\"has any of the words\\",\\"operands\\":1,\\"visibility\\":[\\"message\\",\\"message_offline\\"]}]"
                  />
                </div>
              </div>
              <div
                class="single_operand"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="-1"
                      name="arguments"
                      placeholder=""
                      type="text"
                      value="Red, r"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="categorize_as"
                data-draggable="true"
              >
                categorize as
              </div>
              <div
                class="category"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="36"
                      name="Exit Name"
                      type="text"
                      value="Red"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="fe-x remove_icon"
                data-testid="remove-case-b1f332f3-bdd3-4891-aec5-1843a712dbf1"
              />
            </div>
          </div>
        </div>
        <div
          class="kase case_list_case"
        >
          <div
            class="ele group"
          >
            <div
              class="kase has_any_word"
              data-draggable="true"
            >
              <span
                class="fe-chevrons-expand dnd_icon"
                data-draggable="true"
              />
              <div
                class="choice"
              >
                <div
                  class="small "
                >
                  <temba-select
                    data-testid="temba_select_operator"
                    errors="[]"
                    name="operator"
                    namekey="verboseName"
                    valuekey="type"
                    values="[{\\"type\\":\\"has_any_word\\",\\"verboseName\\":\\"has any of the words\\",\\"operands\\":1,\\"visibility\\":[\\"message\\",\\"message_offline\\"]}]"
                  />
                </div>
              </div>
              <div
                class="single_operand"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="-1"
                      name="arguments"
                      placeholder=""
                      type="text"
                      value="Green, g"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="categorize_as"
                data-draggable="true"
              >
                categorize as
              </div>
              <div
                class="category"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="36"
                      name="Exit Name"
                      type="text"
                      value="Green"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="fe-x remove_icon"
                data-testid="remove-case-061fc171-8b79-4636-b892-bd0ea5aa9b42"
              />
            </div>
          </div>
        </div>
        <div
          class="kase case_list_case"
        >
          <div
            class="ele group"
          >
            <div
              class="kase has_any_word"
              data-draggable="true"
            >
              <span
                class="fe-chevrons-expand dnd_icon"
                data-draggable="true"
              />
              <div
                class="choice"
              >
                <div
                  class="small "
                >
                  <temba-select
                    data-testid="temba_select_operator"
                    errors="[]"
                    name="operator"
                    namekey="verboseName"
                    valuekey="type"
                    values="[{\\"type\\":\\"has_any_word\\",\\"verboseName\\":\\"has any of the words\\",\\"operands\\":1,\\"visibility\\":[\\"message\\",\\"message_offline\\"]}]"
                  />
                </div>
              </div>
              <div
                class="single_operand"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="-1"
                      name="arguments"
                      placeholder=""
                      type="text"
                      value=""
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="categorize_as"
                data-draggable="true"
              >
                categorize as
              </div>
              <div
                class="category"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="36"
                      name="Exit Name"
                      type="text"
                      value=""
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="fe-x remove_icon"
                data-testid="remove-case-1e47a1e1-3c67-4df5-adf1-da542c789adb"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`CaseList updates should remove cases 1`] = `
<body>
  <div>
    <div
      class="case_list_container "
    >
      <div
        class="case_list"
      >
        <div
          class="kase case_list_case"
        >
          <div
            class="ele group"
          >
            <div
              class="kase has_any_word"
              data-draggable="true"
            >
              <span
                class="fe-chevrons-expand dnd_icon"
                data-draggable="true"
              />
              <div
                class="choice"
              >
                <div
                  class="small "
                >
                  <temba-select
                    data-testid="temba_select_operator"
                    errors="[]"
                    name="operator"
                    namekey="verboseName"
                    valuekey="type"
                    values="[{\\"type\\":\\"has_any_word\\",\\"verboseName\\":\\"has any of the words\\",\\"operands\\":1,\\"visibility\\":[\\"message\\",\\"message_offline\\"]}]"
                  />
                </div>
              </div>
              <div
                class="single_operand"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="-1"
                      name="arguments"
                      placeholder=""
                      type="text"
                      value="Green, g"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="categorize_as"
                data-draggable="true"
              >
                categorize as
              </div>
              <div
                class="category"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="36"
                      name="Exit Name"
                      type="text"
                      value="Green"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="fe-x remove_icon"
                data-testid="remove-case-061fc171-8b79-4636-b892-bd0ea5aa9b42"
              />
            </div>
          </div>
        </div>
        <div
          class="kase case_list_case"
        >
          <div
            class="ele group"
          >
            <div
              class="kase has_any_word"
              data-draggable="true"
            >
              <span
                class="fe-chevrons-expand dnd_icon"
                data-draggable="true"
              />
              <div
                class="choice"
              >
                <div
                  class="small "
                >
                  <temba-select
                    data-testid="temba_select_operator"
                    errors="[]"
                    name="operator"
                    namekey="verboseName"
                    valuekey="type"
                    values="[{\\"type\\":\\"has_any_word\\",\\"verboseName\\":\\"has any of the words\\",\\"operands\\":1,\\"visibility\\":[\\"message\\",\\"message_offline\\"]}]"
                  />
                </div>
              </div>
              <div
                class="single_operand"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="-1"
                      name="arguments"
                      placeholder=""
                      type="text"
                      value=""
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="categorize_as"
                data-draggable="true"
              >
                categorize as
              </div>
              <div
                class="category"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="36"
                      name="Exit Name"
                      type="text"
                      value=""
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="fe-x remove_icon"
                data-testid="remove-case-1e47a1e1-3c67-4df5-adf1-da542c789adb"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`CaseList updates should update cases 1`] = `
<body>
  <div>
    <div
      class="case_list_container "
    >
      <div
        class="case_list"
      >
        <div
          class="kase case_list_case"
        >
          <div
            class="ele group"
          >
            <div
              class="kase has_any_word"
              data-draggable="true"
            >
              <span
                class="fe-chevrons-expand dnd_icon"
                data-draggable="true"
              />
              <div
                class="choice"
              >
                <div
                  class="small "
                >
                  <temba-select
                    data-testid="temba_select_operator"
                    errors="[]"
                    name="operator"
                    namekey="verboseName"
                    valuekey="type"
                    values="[{\\"type\\":\\"has_any_word\\",\\"verboseName\\":\\"has any of the words\\",\\"operands\\":1,\\"visibility\\":[\\"message\\",\\"message_offline\\"]}]"
                  />
                </div>
              </div>
              <div
                class="single_operand"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="-1"
                      name="arguments"
                      placeholder=""
                      type="text"
                      value="Purple, p"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="categorize_as"
                data-draggable="true"
              >
                categorize as
              </div>
              <div
                class="category"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="36"
                      name="Exit Name"
                      type="text"
                      value="Red"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="fe-x remove_icon"
                data-testid="remove-case-b1f332f3-bdd3-4891-aec5-1843a712dbf1"
              />
            </div>
          </div>
        </div>
        <div
          class="kase case_list_case"
        >
          <div
            class="ele group"
          >
            <div
              class="kase has_any_word"
              data-draggable="true"
            >
              <span
                class="fe-chevrons-expand dnd_icon"
                data-draggable="true"
              />
              <div
                class="choice"
              >
                <div
                  class="small "
                >
                  <temba-select
                    data-testid="temba_select_operator"
                    errors="[]"
                    name="operator"
                    namekey="verboseName"
                    valuekey="type"
                    values="[{\\"type\\":\\"has_any_word\\",\\"verboseName\\":\\"has any of the words\\",\\"operands\\":1,\\"visibility\\":[\\"message\\",\\"message_offline\\"]}]"
                  />
                </div>
              </div>
              <div
                class="single_operand"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="-1"
                      name="arguments"
                      placeholder=""
                      type="text"
                      value="Green, g"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="categorize_as"
                data-draggable="true"
              >
                categorize as
              </div>
              <div
                class="category"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="36"
                      name="Exit Name"
                      type="text"
                      value="Green"
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="fe-x remove_icon"
                data-testid="remove-case-061fc171-8b79-4636-b892-bd0ea5aa9b42"
              />
            </div>
          </div>
        </div>
        <div
          class="kase case_list_case"
        >
          <div
            class="ele group"
          >
            <div
              class="kase has_any_word"
              data-draggable="true"
            >
              <span
                class="fe-chevrons-expand dnd_icon"
                data-draggable="true"
              />
              <div
                class="choice"
              >
                <div
                  class="small "
                >
                  <temba-select
                    data-testid="temba_select_operator"
                    errors="[]"
                    name="operator"
                    namekey="verboseName"
                    valuekey="type"
                    values="[{\\"type\\":\\"has_any_word\\",\\"verboseName\\":\\"has any of the words\\",\\"operands\\":1,\\"visibility\\":[\\"message\\",\\"message_offline\\"]}]"
                  />
                </div>
              </div>
              <div
                class="single_operand"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="-1"
                      name="arguments"
                      placeholder=""
                      type="text"
                      value=""
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="categorize_as"
                data-draggable="true"
              >
                categorize as
              </div>
              <div
                class="category"
              >
                <div
                  class="ele"
                >
                  <div
                    class="wrapper"
                  >
                    <input
                      class="textinput"
                      data-spec="input"
                      data-testid="input"
                      maxlength="36"
                      name="Exit Name"
                      type="text"
                      value=""
                    />
                    <div
                      class="completion_container hidden"
                      data-spec="completion-options"
                      style="top: 0px; left: 0px;"
                    >
                      <div
                        class="options_wrapper"
                      >
                        <ul
                          class="option_list"
                          data-spec="completion-list"
                        />
                        <div
                          class="help"
                          data-spec="completion-help"
                        >
                          Tab to complete, enter to select
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="fe-x remove_icon"
                data-testid="remove-case-1e47a1e1-3c67-4df5-adf1-da542c789adb"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
